### PR TITLE
DSCO SimpleDropdown - add errorMessage and helperMessage

### PR DESCRIPTION
**Pull request by drizco**
this pr adds `errorMessage`, `helperMessage` and `helperIcon` to the `SimpleDropdown` component, using `TextField` props and styling as a guide.

Screenshot 2024-10-01 at 12 55 20 PM

## Links

[figma](https://www.figma.com/design/88sbry7xnl1wcyrOgbgFtn/Sign-Up-Flow?node-id=4672-2996&node-type=frame&t=MIwrXy68mdu0mGII-0)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/dropdown/simpleDropdown/CHANGELOG.md
Comment (levadadenys) [inline apps/src/componentLibrary/dropdown/simpleDropdown/CHANGELOG.md]: please add a PR link here

File: apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.story.tsx
Comment (levadadenys) [inline apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.story.tsx]: Thanks a lot for adding all of this stories!
Please note that they won't work untill you replace `DisabledDropdown.args` with `WithErrorDropdown.args`, etc for other new stories. Since right now code only updates DisabledDropdown story

File: apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.story.tsx
Comment (levadadenys) [inline apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.story.tsx]: Please replace `DisabledDropdown.args`

File: apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.story.tsx
Comment (levadadenys) [inline apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.story.tsx]: please replace replace `DisabledDropdown.args`

File: apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.tsx
Comment (levadadenys) [inline apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.tsx]: Let's please rename this className to .hasError so that it'll be more sematic and easier to understand

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss
Comment (levadadenys) [inline apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss]: where do we take this value from?

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss
Comment (levadadenys) [inline apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss:43]: Same question about quotes

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss
Comment (levadadenys) [inline apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss:266]: Where do we take these values from?

**Review by levadadenys — COMMENTED** [review]
Thanks a lot for this update!
Can't approve until we have Figma designs for these, however reviewed and left some comments

File: apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.story.tsx
Comment (drizco) [reply apps/src/componentLibrary/dropdown/simpleDropdown/SimpleDropdown.story.tsx]: oof! thanks for catching this 😅

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss
Comment (drizco) [reply apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss]: I got them from TextField. using gap instead of margin-bottom: 0.37rem below

Screenshot 2024-10-02 at 9 33 17 AM

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss
Comment (drizco) [reply apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss:43]: the prettier extension in my ide is formatting these as single quotes due to the setting in prettierrc.js config in /apps

Comment (drizco) [reply apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss:43]: I can save without formatting but it seems intentional to prefer single quotes 🤷

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss
Comment (drizco) [reply apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss:266]: TextField was my guide on these, since they'll be mixed together in the same form in most cases

**Review by levadadenys — COMMENTED** [review]
LGTM so far! Not approving just because need to discuss with Design team the structure (from what I see in figma the styles that you're adding here might be needed for a new type of Dropdown, meanwhile keeping the simpleDropdown)

For now suggest to put it on pause

**Comment by drizco** [comment]
> LGTM so far! Not approving just because need to discuss with Design team the structure (from what I see in figma the styles that you're adding here might be needed for a new type of Dropdown, meanwhile keeping the simpleDropdown)
> 
> For now suggest to put it on pause

<s>I guess it's not such a simple dropdown anymore 😂 

My reason for picking this up now was to unblock my updates to the account settings form. I'd much prefer to use the design system components, but I guess I'll have to shoehorn in the error state elements where I need them and maybe swap in the new dropdown later when this is sorted out

converting this pr to a draft</s>

after slack discussion, these new props for helper/error messages added to SimpleDropdown are approved

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss
Comment (levadadenys) [inline apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss]: Upon discussion with design team, please remove this from PR for now and add minWidth with help of custom `className` please

**Review by levadadenys — APPROVED** [review]
Discussed this with design team, replied with more context in slack.
LGTM, jsut please remove `minWidth` from this PR for now

**Pull request merged**
